src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -134,87 +134,341 @@ private static Exception ConvertOrWidenPrimitivesEnumsAndPointersIfPossible(obje
                 return CreateChangeTypeArgumentException(srcEEType, dstEEType);
             }
 
+            EETypeElementType srcElementType = srcEEType->ElementType;
+
+            void* rawDstValue = null;
+
+            Unsafe.SkipInit(out char charValue);
+            Unsafe.SkipInit(out sbyte sbyteValue);
+            Unsafe.SkipInit(out byte byteValue);
+            Unsafe.SkipInit(out short shortValue);
+            Unsafe.SkipInit(out ushort ushortValue);
+            Unsafe.SkipInit(out int intValue);
+            Unsafe.SkipInit(out uint uintValue);
+            Unsafe.SkipInit(out long longValue);
+            Unsafe.SkipInit(out ulong ulongValue);
+            Unsafe.SkipInit(out float floatValue);
+            Unsafe.SkipInit(out double doubleValue);
+
+            Unsafe.SkipInit(out dstObject);
+
```

Comment:
```suggestion
```
It should be a problem for the JIT to figure out that these do not need to be preinitialized.

Reply:
If I remove these, the C# code will no longer compile. Do you mean I should just declare them normally and explicitly set them to 0, and just rely on the JIT to remove that initialization instead? I wasn't sure how the JIT could be smart enough to figure out that each value is correctly set only via the switch branch that that specific primitive type would go to.

Reply:
Why would it not compile? We are taking unmanaged addresses of these. There should be no problem with taking an address of uninitialized local.

Reply:
Oh, mh I think I got tripped up by some previous version of the code where I tried that. Fixed in 73a2d5d556971f41740bde39eda77a3c698c42c8 🙂